core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1490,6 +1480,9 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
 
     implicit val classTag: ClassTag[A] = Chunk.classTagOf(chunk)
 
+    override val depth: Int =
```

Comment:
It's `O(m)` to retrieve depth, concat depth, etc., where `m` is the depth of the chunk. In addition at some depth one has to worry about stack overflow in calculating the depth.

Maybe worth running benchmarks to ensure we're not regressing on known cases?

Reply:
`depth` is a `val` in all cases where it is not `0` so retrieving `depth` is always `O(1)`. I think stack overflows are more of an issue with the current implementation than this one. Currently we allow chunks of arbitrary depth to be constructed so just indexing into a chunk could cause a stack overflow in degenerate cases. In contrast here we limit the depth to never exceed `MaxDepthBeforeMaterialize`.

Ran the `MixedChunkBenchmarks` and did not observe any regressions.